### ISSUE TITLE
Doxygen 1.9.7 - @ref not resolved in layout usergroup url field

### ISSUE THREAD

**Issue by adepierre** [doxygen/doxygen#10162]
**Describe the bug**
When using a reference to a page in the url field of a usergroup, doxygen does not resolve it in 1.9.7 (worked in 1.9.6). It tries to access /@ref%20\<refname\> instead.

**Expected behavior**
Clicking the usergroup defined tab should redirect to the referenced page.

**To Reproduce**
[doxygen_mre.zip](https://github.com/doxygen/doxygen/files/11927682/doxygen_mre.zip)
Two md pages that are grouped in a "Pages" tab. Clicking "Pages" should redirect to page bar.md directly using the {#Bar} reference in md file.
Works with 1.9.6 doesn't with 1.9.7

**Version**
Working: 1.9.6
Non working: 1.9.7
Both binaries for Windows 10 64 bit downloaded from doxygen website.

**Comment by albert-github** [comment]
When I click on the word "Pages" in doxygen version 1.9.6 it is redirected to html/_bar.html
When I click on the word "Pages" in doxygen version 1.9.7 I get a "File not found" as it tries to go to "html/@ref Bar"
When I click on the word "Pages" in the current master version of doxygen it is redirected again to html/_bar.html

Looks like this is fixed by means of:
```
Commit: 13f5b6fee413d31404cf403a9855b7a9d257bc4f [13f5b6f]
Date: Tuesday, May 23, 2023 8:45:39 PM
issue #10056 [1.9.7] Markdown Header ID ignored in subdirectories
```

I think the problem is fixed but not released.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
